### PR TITLE
Implement str.expandtabs() based on CPython

### PR DESCRIPTION
**Pull request by densmirn**

### CODE REVIEW

**Comment by densmirn** [comment]
Could you please queue the PR for review?

File: numba/unicode.py [hunk 1]
```diff
@@ -675,6 +676,71 @@ def endswith_impl(a, b):
...
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L11519-L11595    # noqa: E501
+@overload_method(types.UnicodeType, 'expandtabs')
+def unicode_expandtabs(data, tabsize=8):
+    """Implements str.expandtabs()"""
+    thety = tabsize
+    if isinstance(tabsize, types.Omitted):
+        thety = tabsize.value
+        # if the type is optional, the concrete type is the captured type
```
[the remark this answers is not in the document]
Comment (densmirn) [reply numba/unicode.py]: Fixed!

File: numba/unicode.py [hunk 2]
```diff
@@ -675,6 +676,71 @@ def endswith_impl(a, b):
...
+        raise TypingError(
+            '"tabsize" must be {}, not {}'.format(accepted, tabsize))
+
+    def expandtabs_impl(data, tabsize=8):
+        length = len(data)
+        j = line_pos = 0
+        found = False
+        for i in range(length):
+            code_point = _get_code_point(data, i)
+            if code_point == 9: # 0x9 '\t'
```
[the remark this answers is not in the document]
Comment (densmirn) [reply numba/unicode.py]: Added such variables.

**Comment by densmirn** [comment]
Could you please explain how can I fix the flake8 issue from CI?

**Comment by stuartarchibald** [comment]
> Could you please explain how can I fix the flake8 issue from CI?

I'd guess it's this line https://github.com/numba/numba/pull/4858/files#diff-a21787256ad779b4c4c5c99915c71025R2 and that master already has an import of `sys` present (https://github.com/numba/numba/blob/9e12ffe8163674b66f59dd9eea153b849c8a0688/numba/unicode.py#L1), so when this is merged into master (as what happens in the CI testing) there's duplication.

**Comment by stuartarchibald** [comment]
Please could you resolve the conflicts by merging master into this branch. Thanks!

**Comment by densmirn** [comment]
Resolved all the merge conflicts.

**Comment by seibert** [comment]
Resolved another merge conflict caused by merging the other string PRs.  Waiting for Azure to confirm this PR is OK, then will merge.

**Comment by densmirn** [comment]
> Resolved another merge conflict caused by merging the other string PRs. Waiting for Azure to confirm this PR is OK, then will merge.

There are no conflicts.

**Pull request merged**